### ISSUE TITLE
Offset current without load

### ISSUE THREAD

**Issue by manutoky** [nDenerserve/SmartPi#95]
All four current sensors give me a reading of -0.1 A without load. This means a power offset of about 20 W. Reading through the forum I have the impression that this a bug regarding the setup of the measurement hardware. Because of this (and because I cannot post on the forum since nobody clears my registration) I opened this issue.
SmartPi 1.0

**Comment by frankrichter** [comment]
The current offset we discussed in the forum recently doesn't affect the power reading (at least in my case).

**Comment by manutoky** [comment]
How is this possible? Power should be calculated using the measurements of current, voltage and phase. This is why I get around 20W of power when reading 0.1A current at 230V.

**Comment by frankrichter** [comment]
Power is calculated internally by the ADE7878 DSP.

**Comment by manutoky** [comment]
This is not a contradiction. ADE7878 only measures currents, voltages and phases. Power will be calculated using these. Offsets in current will result in offsets in power.

**Comment by frankrichter** [comment]
Before I changed the ade7878.go as discussed in the forum my SmartPi displayed -0,07A on all phases, but the power reading was about 0W with small fluctuations. With the changes applied all currents are very near to 0 now, but the power readings didn't change at all. The xIRMSOS calibration only affects the xRMS registers, not the power registers.

Do you use the newest Software on your SmartPi? Then this should be the same for you.

**Comment by frankrichter** [comment]
Btw: I wrote an email to info@enerserve.eu to get my forum account activated.

**Comment by manutoky** [comment]
Yes, I'm using SmartPi 1.0 which I downloaded on saturday. There were no updates using apt-get. I will take a look at the software since there are a couple things I want to change. (Running X.org and chromium processes smashing my raspi for example, update frequency of MQTT too high, etc.). But I should probably discuss those topics in the forum once I have access.

**Comment by manutoky** [comment]
My solution for now is to change the line
`outcome = outcome - 7300`
to
`outcome = outcome - 3200`

**Comment by frankrichter** [comment]
This will not have the same effect, because the xIRMSOS offset are added before the square root is executed.
